enhancements/helm-mediatype.md

```diff
@@ -0,0 +1,320 @@
+---
+title: helm-mediatype
+authors:
+  - "@bowenislandsong"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-May-26
+last-updated: 2020-May-29
+status: provisional
+see-also:
+  - "./enhancements/csvless-bundles.md"
+---
+
+# helm-mediatype
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+ - Can OLM support Chart deprecation from a registry? [Deprecate Charts](https://helm.sh/docs/topics/charts/#deprecating-charts)?
+ - Can OLM support at least parts of the Chart [Hooks](https://helm.sh/docs/topics/charts_hooks/) functionalities such as pre/post-install/upgrade? (OLM does not have operator rollback, delete, and test features.)
+ - Should an operator package allow bundles of mixed media types and will upgrades between them sail smoothly?
+ - How to avoid OLM/Registry from the previous versions to install/add Helm charts?
+ - Can install mode for an operator be default to a value/inferred based on any information in the chart?
+ - Is it safe to assume that all [built-in objects](https://helm.sh/docs/chart_template_guide/builtin_objects/), including capabilities, are evaluated during chart render time and should not affect OLM operations?
+ - What would be a good CLI UX experience for users to putin chart settings/values on subscription/update?
+
+## Summary
+
+This enhancement adds support to accept `helm chart` media type for [Operator Bundle spec](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md), [operator-registry](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/api/apis/operators/v1alpha1/clusterserviceversion_types.go), and [OLM](https://github.com/operator-framework/operator-lifecycle-manager) and enables deployment/upgrading helm-based operators. This enhancement covers the design/UX for opm and OLM, and pipeline for building, adding, and installing helm-based operators that follow the [conventions](https://helm.sh/docs/chart_best_practices/conventions/) of `Helm+v3`.
+
+## Motivation
+Operator authors and cluster admins may find themselves in the split between choosing to support/install operators via OLM or Helm. Helm users hoping to have OLM's operator management experience would have to convert their existing chart into OLM bundle. As many similarities as there are between the two projects, supporting/installing through either mechanism expects different packaging methods and pipeline. By supporting [Helm chart](https://hub.helm.sh/) as a bundle media type, OLM would become approachable to the Helm community without requiring operator authors to explicitly support OLM. It also gives cluster admins the ability to bring Helm-based operators on clusters using OLM.
```

Comment:
Have we thought about what helm users lose from converting their bundle to an OLM bundle? Helm has a concept of revisions, which allow you to track what was deployed to the cluster and when. By converting to an OLM bundle, we do enable manifests to get on the cluster, but lose other helm related metadata. Have we thought about importing helm libraries and providing a more first-class experience for existing helm users?